src/linear_algebra/gram_schmidt_ortho.lean

```diff
@@ -0,0 +1,127 @@
+/-
+Copyright (c) 2022 Jiale Miao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiale Miao, Kevin Buzzard
+-/
+
+import tactic.basic
+import algebra.big_operators.basic
+import analysis.inner_product_space.basic
+import analysis.inner_product_space.projection
+import analysis.normed_space.is_R_or_C
+
+/-!
+# Gram-Schmidt Orthogonalization and Orthonormalization
+
+In this file we introduce Gram-Schmidt Orthogonalization and Orthonormalization
+
+## Main results
+
+- `gram_schmidt_process` : Gram-Schmidt Process
+- `gram_schmidt_process_orthogonal` :
+  the proof that "gram_schmidt_process" produces an orthogonal system of vectors
+- `gram_schmidt_process_normed` :
+  Normalized "Gram-Schmidt" (i.e each vector in this system has unit length)
+- `gram_schmidt_process_orthornormal` :
+  the proof that "gram_schmidt_process_normed" produces an orthornormal system of vectors
+-/
+
+open_locale big_operators
+
+variables (𝕜 : Type*) (E : Type*) [is_R_or_C 𝕜] [inner_product_space 𝕜 E]
+
+/-- Definition of Gram-Schmidt Process -/
+noncomputable def gram_schmidt_process (f : ℕ → E) : ℕ → E
+| n := f n - ∑ i in finset.range n,
+  if h1 : i < n then (orthogonal_projection (𝕜 ∙ (gram_schmidt_process i)) (f n) : E) else f 37
+
+/-- 'gram_schmidt_process_def' gets rid of 'ite' in the definition of gram_schmidt_process -/
+lemma gram_schmidt_process_def (f : ℕ → E) (n : ℕ) :
+gram_schmidt_process 𝕜 E f n = f n - ∑ i in finset.range n,
+(orthogonal_projection (𝕜 ∙ (gram_schmidt_process 𝕜 E f i)) (f n) : E) :=
+begin
+  rw [gram_schmidt_process, sub_right_inj],
+  apply finset.sum_congr rfl,
+  intros x hx,
+  rw finset.mem_range at hx,
+  rw if_pos hx,
+end
```

Comment:
Repeating the earlier suggestion since the rename made github hide it:
```suggestion
noncomputable def gram_schmidt_process (f : ℕ → E) : ℕ → E
| n := f n - ∑ i : fin n, have ↑i < n := i.prop,
  orthogonal_projection (𝕜 ∙ gram_schmidt_process i) (f n)

/-- 'gram_schmidt_process_def' turns the sum over `fin n` into a sum over `ℕ`. -/
lemma gram_schmidt_process_def (f : ℕ → E) (n : ℕ) :
gram_schmidt_process 𝕜 E f n = f n - ∑ i in finset.range n,
  orthogonal_projection (𝕜 ∙ gram_schmidt_process 𝕜 E f i) (f n) :=
begin
  rw gram_schmidt_process,
  congr' 1,
  exact fin.sum_univ_eq_sum_range (λ i,
    (orthogonal_projection (𝕜 ∙ gram_schmidt_process 𝕜 E f i) (f n) : E)) n,
end
```

Reply:
Oh, it works now, I have fixed it, thank you!